### ISSUE TITLE
failed when promote snap using option: --yes

### ISSUE THREAD

**Issue by boukendesho** [snapcore/snapcraft#4312]
### Bug Description

I want to promote a snap without confirmation, so i use --yes option

### To Reproduce

snapcraft promote SNAP_NAME --from-channel latest/edge --to-channel latest/candidate --yes

then i got

Error: 'latest/edge' is not a valid set value for --from-channel when using --yes.

### Environment

Ubuntu 23.04
snapcraft 7.5.2

### snapcraft.yaml

```shell
no
```

### Relevant log output

```shell
no log
```

**Comment by boukendesho** [comment]
snapcraft promote SNAP_NAME --from-channel edge --to-channel candidate --yes

drop the 'latest/' track name also got same error

**Comment by sergiusens** [comment]
Thanks for the report. Is the error confusing? what we are essentially saying is that you shouldn't be auto promoting from edge as it is a moving target

**Comment by boukendesho** [comment]
@sergiusens Yeah a bit confusing to me, so this is a normal behavior by design? I want to automatically promote through cron tasks. Is there any way to do this currently?

**Comment by sergiusens** [comment]
Not currently, but we can consider adding a flag like `--enable-dangerous-promotion`. Since edge is a moving target, there is no set that can be created from `edge` in a reliable way. We could also consider creating sets with version and/or build ids (launchpad and our action-build add these)

**Comment by boukendesho** [comment]
@sergiusens ok, thx

**Comment by krille-chan** [comment]
This god closed, does it mean that `--enable-dangerous-promotion` is now implemented?

**Comment by mr-cal** [comment]
This has been fixed in Snapcraft 9.0.

Anyone using workarounds in a Github workflow like this:

```yaml
env:
    SNAPCRAFT_HAS_TTY: "true"
run: |
  yes | snapcraft promote <snap-name> --from-channel latest/edge --to-channel latest/beta
```

can use the `--yes` flag instead:

```yaml
run: |
  snapcraft promote <snap-name> --from-channel latest/edge --to-channel latest/beta --yes
```
